xarray/backends/zarr.py

```diff
@@ -178,24 +177,24 @@ def _determine_zarr_chunks(enc_chunks, var_chunks, ndim):
     # Here we re-implement this expansion ourselves. That makes the logic of
     # checking chunk compatibility easier
 
-    # this coerces a single int to a tuple but leaves a tuple as is
-    enc_chunks_tuple = tuple(enc_chunks)
-    if len(enc_chunks_tuple) == 1:
-        enc_chunks_tuple = ndim * enc_chunks_tuple
+    if type(enc_chunks) in integer_types:
```

Comment:
Use `isinstance(enc_chunks, integer_types)` .`type(enc_chunks)` is never going to be exactly `integer_types` (which is actually a tuple of types).

(This might by why your tests are failing.)